src/argilla/client/models.py

```diff
@@ -106,10 +115,29 @@ def _none_to_empty_dict(cls, v):
             return {}
         return v
 
-    @validator("id", check_fields=False, always=True)
-    def _none_to_generated_uid64(cls, v):
+    @validator("id", check_fields=False, pre=True, always=True)
+    def _normalize_id(cls, v):
         if v is None:
             return str(uuid.uuid4())
+        if isinstance(v, int):
+            message = (
+                f"Integer ids won't be supported in future versions. We recommend to start using strings instead. "
+                "For datasets already containing integer values we recommend migrating them to avoid deprecation issues."
+                "See https://docs.argilla.io/en/latest/getting_started/installation/configurations"
+                "/database_migrations.html#elasticsearch"
+            )
+            warnings.warn(message, DeprecationWarning)
+            # See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER
+            if v > _JS_MAX_SAFE_INTEGER:
+                message = (
+                    "You've provided a big integer value. Use a string instead, otherwise you may experience some "
+                    "problems using the UI. See "
+                    "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number"
+                    "/MAX_SAFE_INTEGER"
+                )
+                warnings.warn(message, UserWarning)
+        elif not isinstance(v, str):
```

Comment:
This manual type-condition is needed. I tried using `conint` and `constr` in the field definition but I found some weird behavior and unexpected int->str automatic conversions.